service/worker/scanner/history/scavenger.go

```diff
@@ -85,11 +85,8 @@ const (
 	numWorker = 10
 
 	// only clean up history branches that older than this threshold
-	// we double the MaxWorkflowRetentionPeriod to avoid racing condition with history archival.
 	// Our history archiver delete mutable state, and then upload history to blob store and then delete history.
-	// This scanner will face racing condition with archiver because it relys on describe mutable state returning entityNotExist error.
-	// That's why we need to keep MaxWorkflowRetentionPeriod stable and not decreasing all the time.
-	cleanUpThreshold = common.MaxWorkflowRetentionPeriod * 2
+	cleanUpThreshold = 60 * 24 * time.Hour
```

Comment:
I think we should use a larger duration. The reason is, for each history branch, the scavenger will check if they are older than this threshold, if so it will load mutable state, otherwise it will skip. For majority case, this will be a no-op, so loading mutable state is quite costly if we have to load every single mutable state. Make the duration longer will reduce the chance because by the time when it is over the threshold duration, most of history will already be deleted. This also depends on the duration of the life time of a workflow. So I think we may even need to consider make this configurable via dynamic config. The cost of keeping small percent of garbage history is probably less than the cost of scanning the whole history and mutable states.

Reply:
Is it worth to make it configurable and give an arbitrary number?

Reply:
Updated to be configurable and default set to 90 days.